pkg/azurelustreplugin/entrypoint.sh

```diff
@@ -108,15 +111,36 @@ if [[ "${installClientPackages}" == "yes" ]]; then
     echo "deb [arch=amd64] https://packages.microsoft.com/repos/amlfs-${osReleaseCodeName}/ ${osReleaseCodeName} main" | tee /etc/apt/sources.list.d/amlfs.list
     apt-get update
   fi
-  
+
   echo "$(date -u) Installing Lustre client modules: ${pkgName}=${kernelVersion}"
 
-  # grub issue
-  # https://stackoverflow.com/questions/40748363/virtual-machine-apt-get-grub-issue/40751712
-  DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends -o DPkg::options::="--force-confdef" -o DPkg::options::="--force-confold" \
-    ${pkgName}=${kernelVersion}
+  tries=3
+  install_success=false
+  while [[ tries -gt 0 ]]; do
+    # grub issue
+    # https://stackoverflow.com/questions/40748363/virtual-machine-apt-get-grub-issue/40751712
+    if ! DEBIAN_FRONTEND=noninteractive apt install -y --no-install-recommends -o DPkg::options::="--force-confdef" -o DPkg::options::="--force-confold" \
+      ${pkgName}=${kernelVersion}; then
+      echo "$(date -u) Error installing Lustre client modules. Will try removing existing versions"
+      if ! lustre_rmmod; then
```

Comment:
Will this fail if no modules / packages are installed?
Thinking of a case where install failed for some reason, but it wasn't because modules were already installed - this would be a misleading error message.

Reply:
If you had a separate command to check for any installed lustre client packages first, you could then print out the pkg version with a helpful bread crumb.

Reply:
I'll see about having it react better if the modules aren't installed/loaded.

As far as checking for installed packages first, an earlier attempt at this did that (to be able to check if we need to upgrade or not). In a lot of cases, we get back a mishmash of mixed up versions (and the general package without a specified version). If it's useful, I could have it at least print out what apt lists as the installed versions just to give someone debugging a better idea of what's going on.

Reply:
I think you'll benefit from more breadcrumbs if you can get them.